src/components/theme/ThemeProvider.tsx

```diff
@@ -1,31 +1,51 @@
-import React, {PropsWithChildren, useEffect, useMemo, useState} from 'react';
+import React, {PropsWithChildren, useEffect, useLayoutEffect, useMemo, useState} from 'react';
 
-import {DEFAULT_THEME} from './constants';
+import {DEFAULT_LIGHT_THEME, DEFAULT_DARK_THEME, DEFAULT_THEME} from './constants';
 import {ThemeContext} from './ThemeContext';
 import {ThemeValueContext} from './ThemeValueContext';
-import type {Theme} from './types';
+import {ThemeSettings, ThemeSettingsContext} from './ThemeSettingsContext';
+import type {Theme, RealTheme} from './types';
 import {updateBodyClassName} from './updateBodyClassName';
 import {useSystemTheme} from './useSystemTheme';
 
 interface ThemeProviderExternalProps {}
 
 interface ThemeProviderDefaultProps {
     theme: Theme;
+    systemLightTheme: RealTheme;
+    systemDarkTheme: RealTheme;
```

Comment:
Both systemLight and systemDark themes can be set to _any_ theme.

It may sound strange at first, but it allows user to select any theme both for light and dark system themes, just like in github.

<img width="702" alt="Screen Shot 2022-08-31 at 22 40 42" src="https://user-images.githubusercontent.com/168296/187767818-b11a0b33-1c3d-44f7-9604-dbea3dde4d90.png">